File: model/src/main/java/net/artux/pda/model/quest/Stage.java
package net.artux.pda.model.quest;

import java.io.Serializable;
import java.util.HashMap;
import java.util.List;

import lombok.Data;

@Data
public class Stage implements Serializable {

    private Integer id;
    private Integer typeStage;
    private String background;
    private int[] music;
    private String title;
    private String message;
    private Integer typeMessage;
    private List<Text> texts = null;
    private List<Transfer> transfers = null;
    private HashMap<String, List<String>> actions;
    private HashMap<String, String> data;

}

File: model/src/main/java/net/artux/pda/model/quest/Stage.kt
package net.artux.pda.model.quest

import java.io.Serializable

class Stage : Serializable {
    var id: Long = 0
    var typeStage: Int = 0
    var background: String? = ""
    var title: String? = null
    var message: String? = null
    var typeMessage: Int? = null
    var texts: List<Text> = listOf()
    var transfers: List<Transfer> = listOf()
    var actions: HashMap<String, List<String>>? = HashMap()
    var data: HashMap<String, String>? = HashMap()

    fun isNeedSync(): Boolean{
        for (text in texts){
            if (!text.condition.isNullOrEmpty())
                return true
        }
        for (transfer in transfers){
            if (!transfer.condition.isNullOrEmpty())
                return true
        }

        return false
    }
}